language/move-vm/runtime/src/loader.rs

```diff
@@ -1875,3 +1877,45 @@ impl Loader {
         self.type_to_type_layout_impl(ty, 1)
     }
 }
+
+// Public APIs for external uses.
+impl Loader {
+    pub fn get_function_signature(
+        &self,
+        function_name: &IdentStr,
+        module_id: &ModuleId,
+        ty_args: &[TypeTag],
+        move_storage: &impl MoveStorage,
+    ) -> VMResult<(Vec<TypeTag>, Vec<TypeTag>)> {
```

Comment:
Might be worth it to define a struct with `parameter_types` and `return_types` fields for clarity.

Reply:
Also, do we want to return the type parameters and their constraints as well? This function would return the same think for `f<T>(x: T)` and `g<T: drop>(x: t)` even those are different signatures.